core/subscription/extrinsic_event_key_repository.hpp

```diff
@@ -19,6 +21,7 @@ namespace kagome::subscription {
         : logger_{log::createLogger("ExtrinsicEventKeyRepo", "transactions")} {}
 
     ExtrinsicKey add(const primitives::Transaction::Hash &hash) noexcept {
+      std::unique_lock lock{mutex_};
```

Comment:
We could use scoped_lock within the file since we do not need the explicit unlock() interface

Reply:
`scoped_lock` or `lock_guard`?  
I prefer `unique_lock`/`shared_lock`, because it's clear choise between them, and both destructors unlock.